src/host.rs

```diff
@@ -192,6 +192,15 @@ impl<'a> HostAndPort<&'a str> {
     }
 }
 
+impl<S: AsRef<str>> fmt::Display for HostAndPort<S> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        try!(self.host.fmt(f));
```

Comment:
`Display` impls are pretty much never "atomic" in that sense. On the contrary, they are designed to be composable like this. The only way I can think of to make on "atomic" would be to first format to a `String` and then write that, but that would add unnecessary costly memory allocations. A large number of them, for deeply nested `Display::fmt` calls.